src/Sylius/Bundle/ApiBundle/Doctrine/QueryCollectionExtension/CountryCollectionExtension.php

```diff
@@ -51,11 +51,13 @@ public function applyToCollection(
         Assert::keyExists($context, ContextKeys::CHANNEL);
         $channel = $context[ContextKeys::CHANNEL];
 
+        $countriesParameterName = $queryNameGenerator->generateParameterName('countries');
+
         if ($channel->getCountries()->count() > 0) {
             $rootAlias = $queryBuilder->getRootAliases()[0];
             $queryBuilder
-                ->andWhere($rootAlias.'.id in (:countries)' )
-                ->setParameter('countries', $channel->getCountries())
+                ->andWhere($rootAlias.'.id in (:'.$countriesParameterName.')' )
```

Comment:
The rule of thumb in Sylius is to use `sprintf` instead of string concatenation if we are merging anything more than two strings. Can you adjust all calls? 
```suggestion
                ->andWhere(sprintf('%s.id in (:%s)', $rootAlias, $countriesParameterName))
```